src/test/java/com/iopipe/Single.java

```diff
@@ -33,7 +33,16 @@ public Single(Engine __e, String __n)
 			throw new NullPointerException();
 		
 		this.engine = __e;
-		this.basename = __n;
+		
+		// Make sure all names are valid ASCII
```

Comment:
@XerTheSquirrel I'm seeing codePointCount below so maybe it's not ascii already?

Reply:
The code here just involves naming tests for JUnit. The regular expression just changes every other character to an `X`, forcing the test names to be valid ASCII. Java itself internally uses UTF-16, so any character past U+FFFF will require surrogate pairs to represent. The truncation here is valid for any kind of input string including strings limited solely to ASCII characters.

Reply:
This was to make it so that there are no JUnit test names with emoji characters in them (the emoji in question being 🌺 ). The Python test utilizes Unicode for a label (albeit limited) so I figured that adding a test for such emoji characters would be useful.

Reply:
Makes sense to me, thanks for the clarification Stephanie!

Reply:
No problem!